### PR TITLE
Fix `repo:` filter in linked diff search for search-based live preview insight. 

### PR DESCRIPTION
**Pull request by vovakulikov**
Fixes https://github.com/sourcegraph/sourcegraph-search-insights/issues/7

Thanks, @felixfbecker we have a PR with the same fix in the search-based extension codebase about link generation PR https://github.com/sourcegraph/sourcegraph-search-insights/pull/11. This PR adds the same changes and fixes in the live preview creation UI for search-based insight.

### CODE REVIEW

File: client/web/src/insights/core/backend/api/get-search-insight-content/get-search-insight-content.ts [hunk 1]
```diff
@@ -112,9 +112,11 @@ export async function getSearchInsightContent(insight: SearchInsightSettings): P
                 // easier to read (else the date component may be off by one day)
                 const after = formatISO(sub(date, step))
                 const before = formatISO(date)
-                const repoFilters = repos.map(repo => `repo:^${escapeRegExp(repo)}$`).join(' ')
-                const diffQuery = `${repoFilters} type:diff after:${after} before:${before} ${series.query}`
+                const repoFilter = `repo:^(${repos.map(escapeRegExp).join('|')})$`
+                const diffQuery = `${repoFilter} type:diff after:${after} before:${before} ${series.query}`
```
Comment (valerybugakov) [inline client/web/src/insights/core/backend/api/get-search-insight-content/get-search-insight-content.ts:116]: Optional: does it make sense to add a test for this issue to avoid it in the future?

**Review by valerybugakov — APPROVED** [review]

**Review by eseliger — APPROVED** [review]

File: client/web/src/insights/core/backend/api/get-search-insight-content/get-search-insight-content.ts (hunk 1 above)
Comment (vovakulikov) [reply client/web/src/insights/core/backend/api/get-search-insight-content/get-search-insight-content.ts:116]: Yeah. Currently, this logic is implicitly tested by integrations tests. I have some thoughts on how to break down our current approach for API service to test this logic in unit tests and separate fetch calls and mock them in units without jest.mock. 

TLDR; we probably want to move this logic from the API module and have a react custom hook for it then we would be able to test this hook in our unit tests.
